### PR TITLE
Set up the repository and move stuff from apify-sdk and apify-client here

### LINKED ISSUE #1: Initialize this repo
Right now, we have two major Python projects - the [Apify SDK](https://github.com/apify/apify-sdk-python) and the [Apify API client](https://github.com/apify/apify-client-python). They're both using some of the same constants and utilities, and right now they're just copied between the two repositories, because it was the easiest way to get the projects out quickly.

In the long term, we're planning to have more Python libraries and use more shared constants and utilities, so it will be beneficial to have those defined here, in one place, and import them from here in other projects (similar to how we do it with https://github.com/apify/apify-shared-js and JavaScript libraries).

To do that, we need to:
- create a new project on PyPI called `apify-shared`
- copy over the project setup from e.g. https://github.com/apify/apify-client-python to here, and change the library name from `apify-client` to `apify-shared`
- copy the constants and the utilities used in the API client and SDK to here (only the relevant ones, some of them make sense only in the context of one project, so they can stay there)
	- they're typically in the `consts.py` and `_utils.py` files in the project repositories
- publish this on PyPI (the CI copied from the API client should work for this)
- replace the constants in the API client and SDK with those from the shared package (and reexport them for convenience)

There are some things to keep in mind when working on this:
- documentation
	- this project won't have its own documentation, so we can remove all the docs workflows
	- we want to have the constants (not necessarily the utilities yet, because those are all internal) documented in the SDK and API client docs, so we should somehow set up the documentation building there to merge the documentation from the main project to include them
		- it will be a bit ugly, but I think the easiest would be to `git clone` the right tag of this repo when building the project docs, build the documentation for this project, and then merge the resulting API reference JSON in the project's API reference JSON.
- sharing the repo setup across Python projects
	- if we add more and more Python projects, we should somehow share the linter & type checker configs between the repos, so if we e.g. add a linter rule, we don't have to add it to every repo separately
	- this is not necessary for this repo's initialization, just something to start thinking about
- sharing constants between JS and Python
	- there will be a lot of constants that are shared between JS and Python (probably almost all of them)
	- we should start thinking about how to make sure they're the same
		- either we should single-source them and automatically generate the JS and Python packages
		- but probably it would be enough to just write a CI job which checks that they're the same regularly, there won't be that many changes to make single-sourcing worth the effort
	- again, this is not necessary for this repo's initialization, just something to start thinking about

### PR DESCRIPTION
**Pull request by vdusek**
I separated the PR into 3 commits. At first, I just copied the `apify-client-python` repository, and in the next commits, I did the changes to be able to see the changes I made.

Basically regarding the project setup I just removed the docs & test stuff.

The content of `src/` was discussed on Slack with @fnesveda .

Closes #1.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+.PHONY: clean install-dev build publish twine-check lint type-check check-code format check-changelog-entry
+
+# This is default for local testing, but GitHub workflows override it to a higher value in CI
+INTEGRATION_TESTS_CONCURRENCY = 1
+
```
Comment (fnesveda) [inline Makefile:5]: This package won't have any integration tests, so no need for this argument
```suggestion
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  "Operating System :: OS Independent",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Topic :: Software Development :: Libraries",
+]
+dependencies = [
+  "httpx",
+]
```
Comment (fnesveda) [inline pyproject.toml:16]: This package doesn't use `httpx` anywhere, no need for this
```suggestion
```

**Review by fnesveda — CHANGES_REQUESTED** [review]
Cool! Just some small notes.

And you should also copy the unit tests from https://github.com/apify/apify-sdk-python/blob/master/tests/unit/test_utils.py and https://github.com/apify/apify-sdk-python/blob/master/tests/unit/test_consts.py to here.

**Comment by vdusek** [comment]
Thanks, @fnesveda. I fixed the minor issues you mentioned, added unit tests, and also a few lines into README.

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,28 @@
+[flake8]
+filename =
+    ./scripts/*.py,
+    ./src/*.py,
```
Comment (fnesveda) [inline .flake8:4]: Linting of tests won't work without this:
```suggestion
filename =
    ./scripts/*.py,
    ./src/*.py,
    ./tests/*.py
```

File: tests/unit/test_consts.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import itertools
+
+from apify_shared.consts import BOOL_ENV_VARS, DATETIME_ENV_VARS, FLOAT_ENV_VARS, INTEGER_ENV_VARS, STRING_ENV_VARS, ApifyEnvVars
+
+
+class TestConsts:  # noqa: D101
```
Comment (fnesveda) [inline tests/unit/test_consts.py:6]: This `noqa` is not needed (I wonder why `flake8-noqa` didn't report it...)
```suggestion
class TestConsts:
```

File: src/apify_shared/utils.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+ListOrDict = TypeVar('ListOrDict', List, Dict)
+T = TypeVar('T')
+
+
+def filter_out_none_values_recursively(dictionary: Dict) -> Dict:
+    """Return copy of the dictionary, recursively omitting all keys for which values are None."""
+    return cast(dict, filter_out_none_values_recursively_internal(dictionary))
+
+
+def filter_out_none_values_recursively_internal(
```
Comment (fnesveda) [inline src/apify_shared/utils.py:20]: I think this method could be prefixed with `_`, to make it explicit that it's private and internal.

**Review by fnesveda — CHANGES_REQUESTED** [review]
Looks good, last few tiny things I noticed, and then I think it's good to go.

**Comment by vdusek** [comment]
Thanks, @fnesveda, it should be fixed now.

**Pull request merged**